From b9a0df2f1f4923916299459e8d56942dc8c45e70 Mon Sep 17 00:00:00 2001
From: Federico Vaga <federico.vaga@vaga.pv.it>
Date: Sun, 23 Apr 2017 12:22:58 +0200
Subject: [PATCH 133/134] trace-cmd: BUG fix malloc() pointer validation

To reproduce the bug

mkdir /sys/kernel/debug/tracing/instances/test
./trace-cmd show -B test -s -f
  Failed to allocate instance path snapshot

Link: http://lkml.kernel.org/r/20170423102258.21609-6-federico.vaga@vaga.pv.it

Signed-off-by: Federico Vaga <federico.vaga@vaga.pv.it>
Signed-off-by: Steven Rostedt (VMware) <rostedt@goodmis.org>
---
 trace-cmd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/trace-cmd.c b/trace-cmd.c
index f7ea5f205d73..6a023e87c4ef 100644
--- a/trace-cmd.c
+++ b/trace-cmd.c
@@ -619,7 +619,7 @@ int main (int argc, char **argv)
 		if (buffer) {
 			path = malloc(strlen(buffer) + strlen("instances//") +
 				      strlen(file) + 1);
-			if (path)
+			if (!path)
 				die("Failed to allocate instance path %s", file);
 			sprintf(path, "instances/%s/%s", buffer, file);
 			file = path;
-- 
2.5.5

